### PR TITLE
Add `hypre_ParCSRMatrixBlockColSum`

### PR DESCRIPTION
**Pull request by victorapm**
Add a `ParCSRMatrix` function for computing its block column sum.

This function will be used by MGR (GEOS).

Depends on #1022

### CODE REVIEW

File: src/parcsr_mv/par_csr_matop.c [hunk 1]
```diff
@@ -6218,26 +6219,20 @@ hypre_ParCSRMatrixDropSmallEntries( hypre_ParCSRMatrix *A,
       return hypre_error_flag;
    }
 
-   hypre_GpuProfilingPushRange("ParCSRMatrixDropSmallEntries");
```
Comment (oseikuffuor1) [inline src/parcsr_mv/par_csr_matop.c:6221]: @victorapm why is this no longer needed? Was it a bug to be here and not in device code? Similar question in the DiagScaleVector function.

**Review by oseikuffuor1 — APPROVED** [review]
@victorapm I left you one comment. Otherwise this looks good to me. Thanks.

File: src/parcsr_mv/par_csr_matop.c (hunk 1 above)
Comment (victorapm) [reply src/parcsr_mv/par_csr_matop.c:6221]: It's already inside the `Device` function and no longer needed here

**Comment by victorapm** [comment]
Hi all, I updated the new function in this PR to use `hypre_DenseBlockMatrix` type as output. This change is more appropriate for the purpose of the new function (to create new block-interpolation options for MGR).

**Review by ulrikeyang — APPROVED** [review]
This looks fine.

**Pull request merged**